From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sun, 18 Sep 2016 12:56:20 +0200
Subject: [PATCH] {PRAGMA} isdn/eicon: silent -Wsuggest-attribute=format and
 -Wformat-security with pragmas

This allows fixing all formatting errors too.
---
 drivers/isdn/hardware/eicon/capifunc.c  |  15 +++-
 drivers/isdn/hardware/eicon/dadapter.c  |  12 +--
 drivers/isdn/hardware/eicon/debug.c     |   3 +
 drivers/isdn/hardware/eicon/debuglib.c  |   9 ++-
 drivers/isdn/hardware/eicon/debuglib.h  |  59 +++++++--------
 drivers/isdn/hardware/eicon/di.c        |   8 ++
 drivers/isdn/hardware/eicon/diva.c      |   4 +-
 drivers/isdn/hardware/eicon/divasfunc.c |   2 +-
 drivers/isdn/hardware/eicon/divasi.c    |   6 +-
 drivers/isdn/hardware/eicon/divasmain.c |  10 +--
 drivers/isdn/hardware/eicon/idifunc.c   |   2 +-
 drivers/isdn/hardware/eicon/io.c        |   7 +-
 drivers/isdn/hardware/eicon/message.c   | 126 ++++++++++++++++----------------
 drivers/isdn/hardware/eicon/mntfunc.c   |   2 +-
 drivers/isdn/hardware/eicon/os_4bri.c   |  11 ++-
 drivers/isdn/hardware/eicon/os_bri.c    |   4 +-
 drivers/isdn/hardware/eicon/os_pri.c    |  36 ++++-----
 drivers/isdn/hardware/eicon/platform.h  |   4 +-
 drivers/isdn/hardware/eicon/s_4bri.c    |  18 ++---
 drivers/isdn/hardware/eicon/um_idi.c    |  62 ++++++++--------
 20 files changed, 216 insertions(+), 184 deletions(-)

diff --git a/drivers/isdn/hardware/eicon/capifunc.c b/drivers/isdn/hardware/eicon/capifunc.c
index 7a0bdbdd87ea..5b696bf46980 100644
--- a/drivers/isdn/hardware/eicon/capifunc.c
+++ b/drivers/isdn/hardware/eicon/capifunc.c
@@ -62,9 +62,11 @@ extern void DIVA_DIDD_Read(DESCRIPTOR *, int);
 /*
  * debug
  */
-static void no_printf(unsigned char *, ...);
+static __printf(1, 2) void no_printf(const char *, ...);
 #include "debuglib.c"
-static void xlog(char *x, ...)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wsuggest-attribute=format"
+static void xlog(const char *x, ...)
 {
 #ifndef DIVA_NO_DEBUGLIB
 	va_list ap;
@@ -81,6 +83,7 @@ static void xlog(char *x, ...)
 	}
 #endif
 }
+#pragma GCC diagnostic pop
 
 /*
  * info for proc
@@ -103,7 +106,7 @@ static void stop_dbg(void)
 /*
  * dummy debug function
  */
-static void no_printf(unsigned char *x, ...)
+static __printf(1, 2) void no_printf(const char *x, ...)
 {
 }
 
@@ -318,8 +321,11 @@ void sendf(APPL *appl, word command, dword Id, word Number, byte *format, ...)
 			if (myDriverDebugHandle.dbgMask & DL_BLK) {
 				xlog("\x00\x02", &msg, 0x81, length);
 				for (i = 0; i < dlength; i += 256) {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-security"
 					DBG_BLK((((char *)(long)GET_DWORD(&msg.info.data_b3_ind.Data)) + i,
 						 ((dlength - i) < 256) ? (dlength - i) : 256))
+#pragma GCC diagnostic pop
 						if (!(myDriverDebugHandle.dbgMask & DL_PRV0))
 							break; /* not more if not explicitly requested */
 				}
@@ -961,9 +967,12 @@ static u16 diva_send_message(struct capi_ctr *ctrl,
 			for (j = 0; j <
 				     GET_WORD(&msg->info.data_b3_req.Data_Length);
 			     j += 256) {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-security"
 				DBG_BLK((((char *) this->xbuffer_ptr[i]) + j,
 					 ((GET_WORD(&msg->info.data_b3_req.Data_Length) - j) <
 					  256) ? (GET_WORD(&msg->info.data_b3_req.Data_Length) - j) : 256))
+#pragma GCC diagnostic pop
 					if (!(myDriverDebugHandle.dbgMask & DL_PRV0))
 						break;	/* not more if not explicitly requested */
 			}
diff --git a/drivers/isdn/hardware/eicon/dadapter.c b/drivers/isdn/hardware/eicon/dadapter.c
index 168366741301..322b4142e831 100644
--- a/drivers/isdn/hardware/eicon/dadapter.c
+++ b/drivers/isdn/hardware/eicon/dadapter.c
@@ -48,7 +48,7 @@ static dword Adapters = 0; /* Number of adapters */
    Shadow IDI_DIMAINT
    and 'shadow' debug stuff
    -------------------------------------------------------------------------- */
-static void no_printf(unsigned char *format, ...)
+static __printf(1, 2) void no_printf(const char *format, ...)
 {
 #ifdef EBUG
 	va_list ap;
@@ -113,7 +113,7 @@ static int diva_didd_add_descriptor(DESCRIPTOR *d) {
 			MAdapter.request = d->request;
 			dprintf = (DIVA_DI_PRINTF)d->request;
 			diva_notify_adapter_change(&MAdapter, 0); /* Inserted */
-			DBG_TRC(("DIMAINT registered, dprintf=%08x", d->request))
+			DBG_TRC(("DIMAINT registered, dprintf=%p", d->request))
 				} else {
 			DBG_TRC(("DIMAINT removed"))
 				diva_notify_adapter_change(&MAdapter, 1); /* About to remove */
@@ -129,7 +129,7 @@ static int diva_didd_add_descriptor(DESCRIPTOR *d) {
 			Adapters++;
 			diva_os_leave_spin_lock(&didd_spin, &irql, "didd_add");
 			diva_notify_adapter_change(d, 0); /* we have new adapter */
-			DBG_TRC(("Add adapter[%d], request=%08x", (i + 1), d->request))
+			DBG_TRC(("Add adapter[%d], request=%p", (i + 1), d->request))
 				return (i + 1);
 		}
 		diva_os_leave_spin_lock(&didd_spin, &irql, "didd_add");
@@ -159,11 +159,11 @@ static int diva_didd_remove_descriptor(IDI_CALL request) {
 			memset(&HandleTable[i], 0x00, sizeof(HandleTable[0]));
 			Adapters--;
 			diva_os_leave_spin_lock(&didd_spin, &irql, "didd_rm");
-			DBG_TRC(("Remove adapter[%d], request=%08x", (i + 1), request))
+			DBG_TRC(("Remove adapter[%d], request=%p", (i + 1), request))
 				return (0);
 		}
 	}
-	DBG_ERR(("Invalid request=%08x, can't remove adapter", request))
+	DBG_ERR(("Invalid request=%p, can't remove adapter", request))
 		return (-1);
 }
 /* --------------------------------------------------------------------------
@@ -277,7 +277,7 @@ static dword diva_register_adapter_callback(		\
 			NotificationTable[i].callback = callback;
 			NotificationTable[i].context = context;
 			diva_os_leave_spin_lock(&didd_spin, &irql, "didd_nfy_add");
-			DBG_TRC(("Register adapter notification[%d]=%08x", i + 1, callback))
+			DBG_TRC(("Register adapter notification[%d]=%p", i + 1, callback))
 				return (i + 1);
 		}
 		diva_os_leave_spin_lock(&didd_spin, &irql, "didd_nfy_add");
diff --git a/drivers/isdn/hardware/eicon/debug.c b/drivers/isdn/hardware/eicon/debug.c
index c3c090919407..ac65e96ef850 100644
--- a/drivers/isdn/hardware/eicon/debug.c
+++ b/drivers/isdn/hardware/eicon/debug.c
@@ -566,6 +566,8 @@ static void DI_format_locked(unsigned short id,
 	DI_format(1, id, type, format, argument_list);
 }
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wsuggest-attribute=format"
 static void DI_format(int do_lock,
 		      unsigned short id,
 		      int type,
@@ -723,6 +725,7 @@ static void DI_format(int do_lock,
 		diva_os_leave_spin_lock(&dbg_q_lock, &old_irql, "format");
 	}
 }
+#pragma GCC diagnostic pop
 
 /*
   Write driver ID and driver revision to callers buffer
diff --git a/drivers/isdn/hardware/eicon/debuglib.c b/drivers/isdn/hardware/eicon/debuglib.c
index d5b1092a54f0..c2b5924eac61 100644
--- a/drivers/isdn/hardware/eicon/debuglib.c
+++ b/drivers/isdn/hardware/eicon/debuglib.c
@@ -35,7 +35,7 @@ DIVA_DI_PRINTF dprintf = no_printf;
 /*****************************************************************************/
 #define DBG_FUNC(name)							\
 	void								\
-	myDbgPrint_##name(char *format, ...)				\
+	myDbgPrint_##name(const char *format, ...)				\
 	{ va_list ap;							\
 		if (myDriverDebugHandle.dbg_prt)			\
 		{ va_start(ap, format);				\
@@ -100,7 +100,12 @@ DbgRegister(char *drvName, char *drvTag, unsigned long dbgMask)
 /*
  * Try to register debugging via old (and only) interface
  */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-contains-nul"
+#pragma GCC diagnostic ignored "-Wformat-extra-args"
+#pragma GCC diagnostic ignored "-Wformat-security"
 	dprintf("\000\377", &myDriverDebugHandle);
+#pragma GCC diagnostic pop
 	if (myDriverDebugHandle.dbg_prt)
 	{
 		return (1);
@@ -137,7 +142,7 @@ DbgDeregister(void)
 	}
 	memset(&myDriverDebugHandle, 0, sizeof(myDriverDebugHandle));
 }
-void xdi_dbg_xlog(char *x, ...) {
+void xdi_dbg_xlog(const char *x, ...) {
 	va_list ap;
 	va_start(ap, x);
 	if (myDriverDebugHandle.dbg_end &&
diff --git a/drivers/isdn/hardware/eicon/debuglib.h b/drivers/isdn/hardware/eicon/debuglib.h
index 2170de140335..a5f8f47f8269 100644
--- a/drivers/isdn/hardware/eicon/debuglib.h
+++ b/drivers/isdn/hardware/eicon/debuglib.h
@@ -103,36 +103,37 @@
 #define DL_TO_KERNEL    0x40000000
 
 #ifdef DIVA_NO_DEBUGLIB
-#define myDbgPrint_LOG(x...) do { } while (0);
-#define myDbgPrint_FTL(x...) do { } while (0);
-#define myDbgPrint_ERR(x...) do { } while (0);
-#define myDbgPrint_TRC(x...) do { } while (0);
-#define myDbgPrint_MXLOG(x...) do { } while (0);
-#define myDbgPrint_EVL(x...) do { } while (0);
-#define myDbgPrint_REG(x...) do { } while (0);
-#define myDbgPrint_MEM(x...) do { } while (0);
-#define myDbgPrint_SPL(x...) do { } while (0);
-#define myDbgPrint_IRP(x...) do { } while (0);
-#define myDbgPrint_TIM(x...) do { } while (0);
-#define myDbgPrint_BLK(x...) do { } while (0);
-#define myDbgPrint_TAPI(x...) do { } while (0);
-#define myDbgPrint_NDIS(x...) do { } while (0);
-#define myDbgPrint_CONN(x...) do { } while (0);
-#define myDbgPrint_STAT(x...) do { } while (0);
-#define myDbgPrint_SEND(x...) do { } while (0);
-#define myDbgPrint_RECV(x...) do { } while (0);
-#define myDbgPrint_PRV0(x...) do { } while (0);
-#define myDbgPrint_PRV1(x...) do { } while (0);
-#define myDbgPrint_PRV2(x...) do { } while (0);
-#define myDbgPrint_PRV3(x...) do { } while (0);
-#define DBG_TEST(func, args) do { } while (0);
-#define DBG_EVL_ID(args) do { } while (0);
+#define myDbgPrint_LOG(x...) do { } while (0)
+#define myDbgPrint_FTL(x...) do { } while (0)
+#define myDbgPrint_ERR(x...) do { } while (0)
+#define myDbgPrint_TRC(x...) do { } while (0)
+#define myDbgPrint_MXLOG(x...) do { } while (0)
+#define myDbgPrint_EVL(x...) do { } while (0)
+#define myDbgPrint_REG(x...) do { } while (0)
+#define myDbgPrint_MEM(x...) do { } while (0)
+#define myDbgPrint_SPL(x...) do { } while (0)
+#define myDbgPrint_IRP(x...) do { } while (0)
+#define myDbgPrint_TIM(x...) do { } while (0)
+#define myDbgPrint_BLK(x...) do { } while (0)
+#define myDbgPrint_TAPI(x...) do { } while (0)
+#define myDbgPrint_NDIS(x...) do { } while (0)
+#define myDbgPrint_CONN(x...) do { } while (0)
+#define myDbgPrint_STAT(x...) do { } while (0)
+#define myDbgPrint_SEND(x...) do { } while (0)
+#define myDbgPrint_RECV(x...) do { } while (0)
+#define myDbgPrint_PRV0(x...) do { } while (0)
+#define myDbgPrint_PRV1(x...) do { } while (0)
+#define myDbgPrint_PRV2(x...) do { } while (0)
+#define myDbgPrint_PRV3(x...) do { } while (0)
+#define DBG_TEST(func, args) do { } while (0)
+#define DBG_EVL_ID(args) do { } while (0)
 
+#error I want to test the prinf debug functions! and commit the do-while without ';' -- IooNag
 #else /* DIVA_NO_DEBUGLIB */
 /*
  * define low level macros for formatted & raw debugging
  */
-#define DBG_DECL(func) extern void  myDbgPrint_##func(char *, ...);
+#define DBG_DECL(func) extern __printf(1, 2) void myDbgPrint_##func(const char *, ...);
 DBG_DECL(LOG)
 DBG_DECL(FTL)
 DBG_DECL(ERR)
@@ -230,10 +231,10 @@ extern void DbgSetLevel(unsigned long dbgMask);
  */
 typedef struct _DbgHandle_ *pDbgHandle;
 typedef void (*DbgEnd)(pDbgHandle);
-typedef void (*DbgLog)(unsigned short, int, const char *, va_list);
-typedef void (*DbgOld)(unsigned short, const char *, va_list);
+typedef __printf(3, 0) void (*DbgLog)(unsigned short, int, const char *, va_list);
+typedef __printf(2, 0) void (*DbgOld)(unsigned short, const char *, va_list);
 typedef void (*DbgEv)(unsigned short, unsigned long, va_list);
-typedef void (*DbgIrq)(unsigned short, int, const char *, va_list);
+typedef __printf(3, 0) void (*DbgIrq)(unsigned short, int, const char *, va_list);
 typedef struct _DbgHandle_
 { char    Registered; /* driver successfully registered */
 #define DBG_HANDLE_REG_NEW 0x01  /* this (new) structure    */
@@ -317,6 +318,6 @@ typedef struct
    Function used for xlog-style debug
    ------------------------------------------------------------- */
 #define XDI_USE_XLOG 1
-void xdi_dbg_xlog(char *x, ...);
+__printf(1, 2) void xdi_dbg_xlog(const char *x, ...);
 #endif /* DIVA_NO_DEBUGLIB */
 #endif /* __DEBUGLIB_H__ */
diff --git a/drivers/isdn/hardware/eicon/di.c b/drivers/isdn/hardware/eicon/di.c
index cd3fba1add12..3deb7b9e490e 100644
--- a/drivers/isdn/hardware/eicon/di.c
+++ b/drivers/isdn/hardware/eicon/di.c
@@ -726,7 +726,15 @@ static byte isdn_ind(ADAPTER *a,
    active board
    ----------------------------------------------------------- */
 static void xdi_xlog(byte *msg, word code, int length) {
+#pragma GCC diagnostic push
+/* clang warning */
+#pragma GCC diagnostic ignored "-Wformat"
+/* gcc warnings */
+#pragma GCC diagnostic ignored "-Wformat-contains-nul"
+#pragma GCC diagnostic ignored "-Wformat-extra-args"
+#pragma GCC diagnostic ignored "-Wformat-security"
 	xdi_dbg_xlog("\x00\x02", msg, code, length);
+#pragma GCC diagnostic pop
 }
 #endif
 /* -----------------------------------------------------------
diff --git a/drivers/isdn/hardware/eicon/diva.c b/drivers/isdn/hardware/eicon/diva.c
index d91dd580e978..1b9a65a1351f 100644
--- a/drivers/isdn/hardware/eicon/diva.c
+++ b/drivers/isdn/hardware/eicon/diva.c
@@ -397,7 +397,7 @@ void *diva_xdi_open_adapter(void *os_handle, const void __user *src,
 
 	if (length < sizeof(diva_xdi_um_cfg_cmd_t)) {
 		DBG_ERR(("A: A(?) open, msg too small (%d < %d)",
-			 length, sizeof(diva_xdi_um_cfg_cmd_t)))
+			 length, (int)sizeof(diva_xdi_um_cfg_cmd_t)))
 			return NULL;
 	}
 	if ((*cp_fn) (os_handle, &msg, src, sizeof(msg)) <= 0) {
@@ -449,7 +449,7 @@ diva_xdi_write(void *adapter, void *os_handle, const void __user *src,
 	if (length < sizeof(diva_xdi_um_cfg_cmd_t)) {
 		DBG_ERR(("A: A(%d) write, message too small (%d < %d)",
 			 a->controller, length,
-			 sizeof(diva_xdi_um_cfg_cmd_t)))
+			 (int)sizeof(diva_xdi_um_cfg_cmd_t)))
 			return (-3);
 	}
 
diff --git a/drivers/isdn/hardware/eicon/divasfunc.c b/drivers/isdn/hardware/eicon/divasfunc.c
index 4be5f8814777..b9110f50bbbf 100644
--- a/drivers/isdn/hardware/eicon/divasfunc.c
+++ b/drivers/isdn/hardware/eicon/divasfunc.c
@@ -36,7 +36,7 @@ static DESCRIPTOR MAdapter;
 /* --------------------------------------------------------------------------
    MAINT driver connector section
    -------------------------------------------------------------------------- */
-static void no_printf(unsigned char *x, ...)
+static __printf(1, 2) void no_printf(const char *x, ...)
 {
 	/* dummy debug function */
 }
diff --git a/drivers/isdn/hardware/eicon/divasi.c b/drivers/isdn/hardware/eicon/divasi.c
index c61049585cbd..688fdf784f68 100644
--- a/drivers/isdn/hardware/eicon/divasi.c
+++ b/drivers/isdn/hardware/eicon/divasi.c
@@ -497,7 +497,7 @@ static int remove_entity(void *entity)
 		return (0);
 	}
 
-	DBG_TRC(("E(%08x) check remove", entity))
+	DBG_TRC(("E(%p) check remove", entity))
 
 		/*
 		  If adapter not answers on remove request inside of
@@ -521,7 +521,7 @@ static int remove_entity(void *entity)
 		remove_wait_queue(&p_os->close_wait, &wait);
 	}
 
-	DBG_TRC(("E(%08x) start remove", entity))
+	DBG_TRC(("E(%p) start remove", entity))
 	{
 		DECLARE_WAITQUEUE(wait, curtask);
 
@@ -538,7 +538,7 @@ static int remove_entity(void *entity)
 		remove_wait_queue(&p_os->close_wait, &wait);
 	}
 
-	DBG_TRC(("E(%08x) remove complete, aborted:%d", entity,
+	DBG_TRC(("E(%p) remove complete, aborted:%d", entity,
 		 p_os->aborted))
 
 		diva_um_idi_stop_wdog(entity);
diff --git a/drivers/isdn/hardware/eicon/divasmain.c b/drivers/isdn/hardware/eicon/divasmain.c
index 8b7ad4f1ab01..11b45e2be0cf 100644
--- a/drivers/isdn/hardware/eicon/divasmain.c
+++ b/drivers/isdn/hardware/eicon/divasmain.c
@@ -175,7 +175,7 @@ static char *getrev(const char *revision)
 	return rev;
 }
 
-void diva_log_info(unsigned char *format, ...)
+void diva_log_info(const char *format, ...)
 {
 	va_list args;
 	unsigned char line[160];
@@ -232,7 +232,7 @@ unsigned long divasa_get_pci_bar(unsigned char bus, unsigned char func,
 		ret = dev->resource[bar].start;
 	}
 
-	DBG_TRC(("GOT BAR[%d]=%08x", bar, ret));
+	DBG_TRC(("GOT BAR[%d]=%08lx", bar, ret));
 
 	{
 		unsigned long type = (ret & 0x00000001);
@@ -243,7 +243,7 @@ unsigned long divasa_get_pci_bar(unsigned char bus, unsigned char func,
 			DBG_TRC(("  memory"));
 			ret &= PCI_BASE_ADDRESS_MEM_MASK;
 		}
-		DBG_TRC(("  final=%08x", ret));
+		DBG_TRC(("  final=%08lx", ret));
 	}
 
 	return (ret);
@@ -419,7 +419,7 @@ diva_os_register_io_port(void *adapter, int on, unsigned long port,
 {
 	if (on) {
 		if (!request_region(port, length, name)) {
-			DBG_ERR(("A: I/O: can't register port=%08x", port))
+			DBG_ERR(("A: I/O: can't register port=%08lx", port))
 				return (-1);
 		}
 	} else {
@@ -431,7 +431,7 @@ diva_os_register_io_port(void *adapter, int on, unsigned long port,
 void __iomem *divasa_remap_pci_bar(diva_os_xdi_adapter_t *a, int id, unsigned long bar, unsigned long area_length)
 {
 	void __iomem *ret = ioremap(bar, area_length);
-	DBG_TRC(("remap(%08x)->%p", bar, ret));
+	DBG_TRC(("remap(%08lx)->%p", bar, ret));
 	return (ret);
 }
 
diff --git a/drivers/isdn/hardware/eicon/idifunc.c b/drivers/isdn/hardware/eicon/idifunc.c
index fef6586fe5ac..1a2c07665180 100644
--- a/drivers/isdn/hardware/eicon/idifunc.c
+++ b/drivers/isdn/hardware/eicon/idifunc.c
@@ -29,7 +29,7 @@ static dword notify_handle;
 static DESCRIPTOR DAdapter;
 static DESCRIPTOR MAdapter;
 
-static void no_printf(unsigned char *x, ...)
+static __printf(1, 2) void no_printf(const char *x, ...)
 {
 	/* dummy debug function */
 }
diff --git a/drivers/isdn/hardware/eicon/io.c b/drivers/isdn/hardware/eicon/io.c
index 8851ce580c23..cb6bea3d75d8 100644
--- a/drivers/isdn/hardware/eicon/io.c
+++ b/drivers/isdn/hardware/eicon/io.c
@@ -111,8 +111,11 @@ dump_xlog_buffer(PISDN_ADAPTER IoAdapter, Xdesc *xlogDesc)
 				break;
 			}
 			logLen = (dword)(GET_WORD(&Xlog[logOut]) - (logOut * sizeof(*Xlog)));
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-security"
 			DBG_FTL_MXLOG(((char *)&Xlog[logOut + 1], (dword)(logLen - 2)))
 				logOut = (GET_WORD(&Xlog[logOut]) + 1) / sizeof(*Xlog);
+#pragma GCC diagnostic pop
 		}
 	DBG_FTL(("%s: ***************** end of XLOG *****************",
 		 &IoAdapter->Name[0]))
@@ -313,7 +316,7 @@ void request(PISDN_ADAPTER IoAdapter, ENTITY *e)
 			if (IoAdapter)
 			{
 				syncReq->GetSerial.serial = IoAdapter->serialNo;
-				DBG_TRC(("xdi: Adapter %d / SerialNo %ld",
+				DBG_TRC(("xdi: Adapter %d / SerialNo %d",
 					 IoAdapter->ANum, IoAdapter->serialNo))
 					return;
 			}
@@ -323,7 +326,7 @@ void request(PISDN_ADAPTER IoAdapter, ENTITY *e)
 			if (IoAdapter)
 			{
 				syncReq->GetCardType.cardtype = IoAdapter->cardType;
-				DBG_TRC(("xdi: Adapter %d / CardType %ld",
+				DBG_TRC(("xdi: Adapter %d / CardType %d",
 					 IoAdapter->ANum, IoAdapter->cardType))
 					return;
 			}
diff --git a/drivers/isdn/hardware/eicon/message.c b/drivers/isdn/hardware/eicon/message.c
index 3e3c2b858b1c..5ed6477e82f1 100644
--- a/drivers/isdn/hardware/eicon/message.c
+++ b/drivers/isdn/hardware/eicon/message.c
@@ -784,7 +784,7 @@ static word get_ncci(PLCI *plci, byte ch, word force_ncci)
 	if (!ch || a->ch_ncci[ch])
 	{
 		ncci_mapping_bug++;
-		dbug(1, dprintf("NCCI mapping exists %ld %02x %02x %02x-%02x",
+		dbug(1, dprintf("NCCI mapping exists %u %02x %02x %02x-%02x",
 				ncci_mapping_bug, ch, force_ncci, a->ncci_ch[a->ch_ncci[ch]], a->ch_ncci[ch]));
 		ncci = ch;
 	}
@@ -821,12 +821,12 @@ static word get_ncci(PLCI *plci, byte ch, word force_ncci)
 					} while ((i < MAX_NL_CHANNEL + 1) && (j < MAX_NCCI + 1));
 					if (i < MAX_NL_CHANNEL + 1)
 					{
-						dbug(1, dprintf("NCCI mapping overflow %ld %02x %02x %02x-%02x-%02x",
+						dbug(1, dprintf("NCCI mapping overflow %u %02x %02x %02x-%02x-%02x",
 								ncci_mapping_bug, ch, force_ncci, i, k, j));
 					}
 					else
 					{
-						dbug(1, dprintf("NCCI mapping overflow %ld %02x %02x",
+						dbug(1, dprintf("NCCI mapping overflow %u %02x %02x",
 								ncci_mapping_bug, ch, force_ncci));
 					}
 					ncci = ch;
@@ -842,7 +842,7 @@ static word get_ncci(PLCI *plci, byte ch, word force_ncci)
 		}
 		a->ncci_ch[ncci] = ch;
 		a->ch_ncci[ch] = (byte) ncci;
-		dbug(1, dprintf("NCCI mapping established %ld %02x %02x %02x-%02x",
+		dbug(1, dprintf("NCCI mapping established %u %02x %02x %02x-%02x",
 				ncci_mapping_bug, ch, force_ncci, ch, ncci));
 	}
 	return (ncci);
@@ -865,7 +865,7 @@ static void ncci_free_receive_buffers(PLCI *plci, word ncci)
 			if (!plci->appl)
 			{
 				ncci_mapping_bug++;
-				dbug(1, dprintf("NCCI mapping appl expected %ld %08lx",
+				dbug(1, dprintf("NCCI mapping appl expected %u %08x",
 						ncci_mapping_bug, Id));
 			}
 			else
@@ -892,7 +892,7 @@ static void ncci_free_receive_buffers(PLCI *plci, word ncci)
 				if (!plci->appl)
 				{
 					ncci_mapping_bug++;
-					dbug(1, dprintf("NCCI mapping no appl %ld %08lx",
+					dbug(1, dprintf("NCCI mapping no appl %u %08x",
 							ncci_mapping_bug, Id));
 				}
 				else
@@ -956,13 +956,13 @@ static void ncci_remove(PLCI *plci, word ncci, byte preserve_ncci)
 		if (a->ncci_plci[ncci] != plci->Id)
 		{
 			ncci_mapping_bug++;
-			dbug(1, dprintf("NCCI mapping doesn't exist %ld %08lx %02x",
+			dbug(1, dprintf("NCCI mapping doesn't exist %u %08x %02x",
 					ncci_mapping_bug, Id, preserve_ncci));
 		}
 		else
 		{
 			cleanup_ncci_data(plci, ncci);
-			dbug(1, dprintf("NCCI mapping released %ld %08lx %02x %02x-%02x",
+			dbug(1, dprintf("NCCI mapping released %u %08x %02x %02x-%02x",
 					ncci_mapping_bug, Id, preserve_ncci, a->ncci_ch[ncci], ncci));
 			a->ch_ncci[a->ncci_ch[ncci]] = 0;
 			if (!preserve_ncci)
@@ -992,7 +992,7 @@ static void ncci_remove(PLCI *plci, word ncci, byte preserve_ncci)
 			if (a->ncci_plci[ncci] == plci->Id)
 			{
 				cleanup_ncci_data(plci, ncci);
-				dbug(1, dprintf("NCCI mapping released %ld %08lx %02x %02x-%02x",
+				dbug(1, dprintf("NCCI mapping released %u %08x %02x %02x-%02x",
 						ncci_mapping_bug, Id, preserve_ncci, a->ncci_ch[ncci], ncci));
 				a->ch_ncci[a->ncci_ch[ncci]] = 0;
 				if (!preserve_ncci)
@@ -1667,7 +1667,7 @@ static byte listen_req(dword Id, word Number, DIVA_CAPI_ADAPTER *a,
 		Info = 0;
 		a->Info_Mask[appl->Id - 1] = GET_DWORD(parms[0].info);
 		a->CIP_Mask[appl->Id - 1] = GET_DWORD(parms[1].info);
-		dbug(1, dprintf("CIP_MASK=0x%lx", GET_DWORD(parms[1].info)));
+		dbug(1, dprintf("CIP_MASK=0x%x", GET_DWORD(parms[1].info)));
 		if (a->Info_Mask[appl->Id - 1] & 0x200) { /* early B3 connect provides */
 			a->Info_Mask[appl->Id - 1] |=  0x10;   /* call progression infos    */
 		}
@@ -4132,7 +4132,7 @@ static void control_rc(PLCI *plci, byte req, byte rc, byte ch, byte global_req,
 		Id = (((dword)(ncci ? ncci : ch)) << 16) | ((word)plci->Id << 8) | a->Id;
 		if (plci->tel && plci->SuppState != CALL_HELD) Id |= EXT_CONTROLLER;
 		Number = plci->number;
-		dbug(1, dprintf("Contr_RC-Id=%08lx,plci=%x,tel=%x, entity=0x%x, command=0x%x, int_command=0x%x", Id, plci->Id, plci->tel, plci->Sig.Id, plci->command, plci->internal_command));
+		dbug(1, dprintf("Contr_RC-Id=%08x,plci=%x,tel=%x, entity=0x%x, command=0x%x, int_command=0x%x", Id, plci->Id, plci->tel, plci->Sig.Id, plci->command, plci->internal_command));
 		dbug(1, dprintf("channels=0x%x", plci->channels));
 		if (plci_remove_check(plci))
 			return;
@@ -4895,7 +4895,7 @@ static void sig_ind(PLCI *plci)
 		return;
 	}
 	if (plci->tel && plci->SuppState != CALL_HELD) Id |= EXT_CONTROLLER;
-	dbug(1, dprintf("SigInd-Id=%08lx,plci=%x,tel=%x,state=0x%x,channels=%d,Discflowcl=%d",
+	dbug(1, dprintf("SigInd-Id=%08x,plci=%x,tel=%x,state=0x%x,channels=%d,Discflowcl=%d",
 			Id, plci->Id, plci->tel, plci->State, plci->channels, plci->hangup_flow_ctrl_timer));
 	if (plci->Sig.Ind == CALL_HOLD_ACK && plci->channels)
 	{
@@ -4997,7 +4997,7 @@ static void sig_ind(PLCI *plci)
 		}
 		if (esc_profile[0])
 		{
-			dbug(1, dprintf("[%06x] CardProfile: %lx %lx %lx %lx %lx",
+			dbug(1, dprintf("[%06x] CardProfile: %x %x %x %x %x",
 					UnMapController(a->Id), GET_DWORD(&esc_profile[6]),
 					GET_DWORD(&esc_profile[10]), GET_DWORD(&esc_profile[14]),
 					GET_DWORD(&esc_profile[18]), GET_DWORD(&esc_profile[46])));
@@ -5074,7 +5074,7 @@ static void sig_ind(PLCI *plci)
 			a->profile.Global_Options |= GL_DTMF_SUPPORTED;
 		}
 		a->manufacturer_features &= ~MANUFACTURER_FEATURE_OOB_CHANNEL;
-		dbug(1, dprintf("[%06x] Profile: %lx %lx %lx %lx %lx",
+		dbug(1, dprintf("[%06x] Profile: %x %x %x %x %x",
 				UnMapController(a->Id), a->profile.Global_Options,
 				a->profile.B1_Protocols, a->profile.B2_Protocols,
 				a->profile.B3_Protocols, a->manufacturer_features));
@@ -5691,7 +5691,7 @@ static void sig_ind(PLCI *plci)
 		}
 		cip = find_cip(a, parms[4], parms[6]);
 		cip_mask = 1L << cip;
-		dbug(1, dprintf("cip=%d,cip_mask=%lx", cip, cip_mask));
+		dbug(1, dprintf("cip=%d,cip_mask=%x", cip, cip_mask));
 		clear_c_ind_mask(plci);
 		if (!remove_started && !a->adapter_disabled)
 		{
@@ -5702,7 +5702,7 @@ static void sig_ind(PLCI *plci)
 				    && (a->CIP_Mask[i] & 1 || a->CIP_Mask[i] & cip_mask)
 				    && CPN_filter_ok(parms[0], a, i)
 				    && test_group_ind_mask_bit(plci, i)) {
-					dbug(1, dprintf("storedcip_mask[%d]=0x%lx", i, a->CIP_Mask[i]));
+					dbug(1, dprintf("storedcip_mask[%d]=0x%x", i, a->CIP_Mask[i]));
 					set_c_ind_mask_bit(plci, i);
 					dump_c_ind_mask(plci);
 					plci->State = INC_CON_PENDING;
@@ -6570,7 +6570,7 @@ static void nl_ind(PLCI *plci)
 	Id = (((dword)(ncci ? ncci : ch)) << 16) | (((word) plci->Id) << 8) | a->Id;
 	if (plci->tel) Id |= EXT_CONTROLLER;
 	APPLptr = plci->appl;
-	dbug(1, dprintf("NL_IND-Id(NL:0x%x)=0x%08lx,plci=%x,tel=%x,state=0x%x,ch=0x%x,chs=%d,Ind=%x",
+	dbug(1, dprintf("NL_IND-Id(NL:0x%x)=0x%08x,plci=%x,tel=%x,state=0x%x,ch=0x%x,chs=%d,Ind=%x",
 			plci->NL.Id, Id, plci->Id, plci->tel, plci->State, ch, plci->channels, plci->NL.Ind & 0x0f));
 
 	/* in the case if no connect_active_Ind was sent to the appl we wait for */
@@ -6986,7 +6986,7 @@ static void nl_ind(PLCI *plci)
 			ncci = get_ncci(plci, ch, 0);
 			Id = (Id & 0xffff) | (((dword) ncci) << 16);
 		}
-		dbug(1, dprintf("N_CONNECT: ch=%d state=%d plci=%lx plci_Id=%lx plci_State=%d",
+		dbug(1, dprintf("N_CONNECT: ch=%d state=%d plci=%x plci_Id=%x plci_State=%d",
 				ch, a->ncci_state[ncci], a->ncci_plci[ncci], plci->Id, plci->State));
 
 		msg = _CONNECT_B3_I;
@@ -9589,7 +9589,7 @@ static void dtmf_enable_receiver(PLCI *plci, byte enable_mask)
 {
 	word min_digit_duration, min_gap_duration;
 
-	dbug(1, dprintf("[%06lx] %s,%d: dtmf_enable_receiver %02x",
+	dbug(1, dprintf("[%06x] %s,%d: dtmf_enable_receiver %02x",
 			(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 			(char *)(FILE_), __LINE__, enable_mask));
 
@@ -9627,7 +9627,7 @@ static void dtmf_send_digits(PLCI *plci, byte *digit_buffer, word digit_count)
 {
 	word w, i;
 
-	dbug(1, dprintf("[%06lx] %s,%d: dtmf_send_digits %d",
+	dbug(1, dprintf("[%06x] %s,%d: dtmf_send_digits %d",
 			(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 			(char *)(FILE_), __LINE__, digit_count));
 
@@ -9659,7 +9659,7 @@ static void dtmf_send_digits(PLCI *plci, byte *digit_buffer, word digit_count)
 static void dtmf_rec_clear_config(PLCI *plci)
 {
 
-	dbug(1, dprintf("[%06lx] %s,%d: dtmf_rec_clear_config",
+	dbug(1, dprintf("[%06x] %s,%d: dtmf_rec_clear_config",
 			(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 			(char *)(FILE_), __LINE__));
 
@@ -9675,7 +9675,7 @@ static void dtmf_rec_clear_config(PLCI *plci)
 static void dtmf_send_clear_config(PLCI *plci)
 {
 
-	dbug(1, dprintf("[%06lx] %s,%d: dtmf_send_clear_config",
+	dbug(1, dprintf("[%06x] %s,%d: dtmf_send_clear_config",
 			(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 			(char *)(FILE_), __LINE__));
 
@@ -10187,7 +10187,7 @@ static void dtmf_parameter_write(PLCI *plci)
 	word i;
 	byte parameter_buffer[DTMF_PARAMETER_BUFFER_SIZE + 2];
 
-	dbug(1, dprintf("[%06lx] %s,%d: dtmf_parameter_write",
+	dbug(1, dprintf("[%06x] %s,%d: dtmf_parameter_write",
 			(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 			(char *)(FILE_), __LINE__));
 
@@ -10204,7 +10204,7 @@ static void dtmf_parameter_write(PLCI *plci)
 static void dtmf_parameter_clear_config(PLCI *plci)
 {
 
-	dbug(1, dprintf("[%06lx] %s,%d: dtmf_parameter_clear_config",
+	dbug(1, dprintf("[%06x] %s,%d: dtmf_parameter_clear_config",
 			(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 			(char *)(FILE_), __LINE__));
 
@@ -10422,7 +10422,7 @@ static void mixer_set_bchannel_id_esc(PLCI *plci, byte bchannel_id)
 					}
 					splci->li_bchannel_id = 3 - plci->li_bchannel_id;
 					li_config_table[a->li_base + (2 - plci->li_bchannel_id)].plci = splci;
-					dbug(1, dprintf("[%06lx] %s,%d: adv_voice_set_bchannel_id_esc %d",
+					dbug(1, dprintf("[%06x] %s,%d: adv_voice_set_bchannel_id_esc %d",
 							(dword)((splci->Id << 8) | UnMapController(splci->adapter->Id)),
 							(char *)(FILE_), __LINE__, splci->li_bchannel_id));
 				}
@@ -10436,7 +10436,7 @@ static void mixer_set_bchannel_id_esc(PLCI *plci, byte bchannel_id)
 	{
 		mixer_clear_config(plci);
 	}
-	dbug(1, dprintf("[%06lx] %s,%d: mixer_set_bchannel_id_esc %d %d",
+	dbug(1, dprintf("[%06x] %s,%d: mixer_set_bchannel_id_esc %d %d",
 			(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 			(char *)(FILE_), __LINE__, bchannel_id, plci->li_bchannel_id));
 }
@@ -10480,7 +10480,7 @@ static void mixer_set_bchannel_id(PLCI *plci, byte *chi)
 						}
 						splci->li_bchannel_id = 3 - plci->li_bchannel_id;
 						li_config_table[a->li_base + (2 - plci->li_bchannel_id)].plci = splci;
-						dbug(1, dprintf("[%06lx] %s,%d: adv_voice_set_bchannel_id %d",
+						dbug(1, dprintf("[%06x] %s,%d: adv_voice_set_bchannel_id %d",
 								(dword)((splci->Id << 8) | UnMapController(splci->adapter->Id)),
 								(char *)(FILE_), __LINE__, splci->li_bchannel_id));
 					}
@@ -10495,7 +10495,7 @@ static void mixer_set_bchannel_id(PLCI *plci, byte *chi)
 	{
 		mixer_clear_config(plci);
 	}
-	dbug(1, dprintf("[%06lx] %s,%d: mixer_set_bchannel_id %02x %d",
+	dbug(1, dprintf("[%06x] %s,%d: mixer_set_bchannel_id %02x %d",
 			(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 			(char *)(FILE_), __LINE__, ch, plci->li_bchannel_id));
 }
@@ -10509,7 +10509,7 @@ static void mixer_calculate_coefs(DIVA_CAPI_ADAPTER *a)
 	char *p;
 	char hex_line[2 * MIXER_MAX_DUMP_CHANNELS + MIXER_MAX_DUMP_CHANNELS / 8 + 4];
 
-	dbug(1, dprintf("[%06lx] %s,%d: mixer_calculate_coefs",
+	dbug(1, dprintf("[%06x] %s,%d: mixer_calculate_coefs",
 			(dword)(UnMapController(a->Id)), (char *)(FILE_), __LINE__));
 
 	for (i = 0; i < li_total_channels; i++)
@@ -10696,7 +10696,7 @@ static void mixer_calculate_coefs(DIVA_CAPI_ADAPTER *a)
 		p = hex_byte_pack(p, li_config_table[j].curchnl);
 	}
 	*p = '\0';
-	dbug(1, dprintf("[%06lx] CURRENT %s",
+	dbug(1, dprintf("[%06x] CURRENT %s",
 			(dword)(UnMapController(a->Id)), (char *)hex_line));
 	p = hex_line;
 	for (j = 0; j < n; j++)
@@ -10706,7 +10706,7 @@ static void mixer_calculate_coefs(DIVA_CAPI_ADAPTER *a)
 		p = hex_byte_pack(p, li_config_table[j].channel);
 	}
 	*p = '\0';
-	dbug(1, dprintf("[%06lx] CHANNEL %s",
+	dbug(1, dprintf("[%06x] CHANNEL %s",
 			(dword)(UnMapController(a->Id)), (char *)hex_line));
 	p = hex_line;
 	for (j = 0; j < n; j++)
@@ -10716,7 +10716,7 @@ static void mixer_calculate_coefs(DIVA_CAPI_ADAPTER *a)
 		p = hex_byte_pack(p, li_config_table[j].chflags);
 	}
 	*p = '\0';
-	dbug(1, dprintf("[%06lx] CHFLAG  %s",
+	dbug(1, dprintf("[%06x] CHFLAG  %s",
 			(dword)(UnMapController(a->Id)), (char *)hex_line));
 	for (i = 0; i < n; i++)
 	{
@@ -10728,7 +10728,7 @@ static void mixer_calculate_coefs(DIVA_CAPI_ADAPTER *a)
 			p = hex_byte_pack(p, li_config_table[i].flag_table[j]);
 		}
 		*p = '\0';
-		dbug(1, dprintf("[%06lx] FLAG[%02x]%s",
+		dbug(1, dprintf("[%06x] FLAG[%02x]%s",
 				(dword)(UnMapController(a->Id)), i, (char *)hex_line));
 	}
 	for (i = 0; i < n; i++)
@@ -10741,7 +10741,7 @@ static void mixer_calculate_coefs(DIVA_CAPI_ADAPTER *a)
 			p = hex_byte_pack(p, li_config_table[i].coef_table[j]);
 		}
 		*p = '\0';
-		dbug(1, dprintf("[%06lx] COEF[%02x]%s",
+		dbug(1, dprintf("[%06x] COEF[%02x]%s",
 				(dword)(UnMapController(a->Id)), i, (char *)hex_line));
 	}
 }
@@ -10865,7 +10865,7 @@ static void xconnect_query_addresses(PLCI *plci)
 	word w, ch;
 	byte *p;
 
-	dbug(1, dprintf("[%06lx] %s,%d: xconnect_query_addresses",
+	dbug(1, dprintf("[%06x] %s,%d: xconnect_query_addresses",
 			(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 			(char *)(FILE_), __LINE__));
 
@@ -10899,7 +10899,7 @@ static void xconnect_query_addresses(PLCI *plci)
 static void xconnect_write_coefs(PLCI *plci, word internal_command)
 {
 
-	dbug(1, dprintf("[%06lx] %s,%d: xconnect_write_coefs %04x",
+	dbug(1, dprintf("[%06x] %s,%d: xconnect_write_coefs %04x",
 			(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 			(char *)(FILE_), __LINE__, internal_command));
 
@@ -10917,14 +10917,14 @@ static byte xconnect_write_coefs_process(dword Id, PLCI *plci, byte Rc)
 	struct xconnect_transfer_address_s   *transfer_address;
 	byte ch_map[MIXER_CHANNELS_BRI];
 
-	dbug(1, dprintf("[%06x] %s,%d: xconnect_write_coefs_process %02x %d",
+	dbug(1, dprintf("[%06lx] %s,%d: xconnect_write_coefs_process %02x %d",
 			UnMapId(Id), (char *)(FILE_), __LINE__, Rc, plci->li_write_channel));
 
 	a = plci->adapter;
 	if ((plci->li_bchannel_id == 0)
 	    || (li_config_table[a->li_base + (plci->li_bchannel_id - 1)].plci != plci))
 	{
-		dbug(1, dprintf("[%06x] %s,%d: Channel id wiped out",
+		dbug(1, dprintf("[%06lx] %s,%d: Channel id wiped out",
 				UnMapId(Id), (char *)(FILE_), __LINE__));
 		return (true);
 	}
@@ -11253,7 +11253,7 @@ static void mixer_notify_update(PLCI *plci, byte others)
 	PLCI *notify_plci;
 	byte msg[sizeof(CAPI_MSG_HEADER) + 6];
 
-	dbug(1, dprintf("[%06lx] %s,%d: mixer_notify_update %d",
+	dbug(1, dprintf("[%06x] %s,%d: mixer_notify_update %d",
 			(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 			(char *)(FILE_), __LINE__, others));
 
@@ -11305,7 +11305,7 @@ static void mixer_notify_update(PLCI *plci, byte others)
 				{
 					if (w != 0)
 					{
-						dbug(1, dprintf("[%06lx] %s,%d: Interconnect notify failed %06x %d",
+						dbug(1, dprintf("[%06x] %s,%d: Interconnect notify failed %06x %d",
 								(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 								(char *)(FILE_), __LINE__,
 								(dword)((notify_plci->Id << 8) | UnMapController(notify_plci->adapter->Id)), w));
@@ -11325,7 +11325,7 @@ static void mixer_clear_config(PLCI *plci)
 	DIVA_CAPI_ADAPTER *a;
 	word i, j;
 
-	dbug(1, dprintf("[%06lx] %s,%d: mixer_clear_config",
+	dbug(1, dprintf("[%06x] %s,%d: mixer_clear_config",
 			(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 			(char *)(FILE_), __LINE__));
 
@@ -11607,7 +11607,7 @@ static void mixer_command(dword Id, PLCI *plci, byte Rc)
 	if ((plci->li_bchannel_id == 0)
 	    || (li_config_table[a->li_base + (plci->li_bchannel_id - 1)].plci != plci))
 	{
-		dbug(1, dprintf("[%06x] %s,%d: Channel id wiped out %d",
+		dbug(1, dprintf("[%06lx] %s,%d: Channel id wiped out %d",
 				UnMapId(Id), (char *)(FILE_), __LINE__, (int)(plci->li_bchannel_id)));
 	}
 	else
@@ -11925,7 +11925,7 @@ static PLCI *li_check_plci_b(dword Id, PLCI *plci,
 	    || (((plci_b_id >> 8) & 0xff) == 0)
 	    || (((plci_b_id >> 8) & 0xff) > adapter[ctlr_b - 1].max_plci))
 	{
-		dbug(1, dprintf("[%06lx] %s,%d: LI invalid second PLCI %08lx",
+		dbug(1, dprintf("[%06lx] %s,%d: LI invalid second PLCI %08x",
 				UnMapId(Id), (char *)(FILE_), __LINE__, plci_b_id));
 		PUT_WORD(p_result, _WRONG_IDENTIFIER);
 		return (NULL);
@@ -11935,7 +11935,7 @@ static PLCI *li_check_plci_b(dword Id, PLCI *plci,
 	    || !plci_b->NL.Id || plci_b->nl_remove_id
 	    || (plci_b->li_bchannel_id == 0))
 	{
-		dbug(1, dprintf("[%06lx] %s,%d: LI peer in wrong state %08lx",
+		dbug(1, dprintf("[%06lx] %s,%d: LI peer in wrong state %08x",
 				UnMapId(Id), (char *)(FILE_), __LINE__, plci_b_id));
 		PUT_WORD(p_result, _REQUEST_NOT_ALLOWED_IN_THIS_STATE);
 		return (NULL);
@@ -11946,7 +11946,7 @@ static PLCI *li_check_plci_b(dword Id, PLCI *plci,
 	    && (!(plci->adapter->manufacturer_features & MANUFACTURER_FEATURE_XCONNECT)
 		|| !(plci_b->adapter->manufacturer_features & MANUFACTURER_FEATURE_XCONNECT)))
 	{
-		dbug(1, dprintf("[%06lx] %s,%d: LI not on same ctrl %08lx",
+		dbug(1, dprintf("[%06lx] %s,%d: LI not on same ctrl %08x",
 				UnMapId(Id), (char *)(FILE_), __LINE__, plci_b_id));
 		PUT_WORD(p_result, _WRONG_IDENTIFIER);
 		return (NULL);
@@ -11988,7 +11988,7 @@ static PLCI *li2_check_plci_b(dword Id, PLCI *plci,
 	    || (((plci_b_id >> 8) & 0xff) == 0)
 	    || (((plci_b_id >> 8) & 0xff) > adapter[ctlr_b - 1].max_plci))
 	{
-		dbug(1, dprintf("[%06lx] %s,%d: LI invalid second PLCI %08lx",
+		dbug(1, dprintf("[%06lx] %s,%d: LI invalid second PLCI %08x",
 				UnMapId(Id), (char *)(FILE_), __LINE__, plci_b_id));
 		PUT_WORD(p_result, _WRONG_IDENTIFIER);
 		return (NULL);
@@ -11999,7 +11999,7 @@ static PLCI *li2_check_plci_b(dword Id, PLCI *plci,
 	    || (plci_b->li_bchannel_id == 0)
 	    || (li_config_table[plci_b->adapter->li_base + (plci_b->li_bchannel_id - 1)].plci != plci_b))
 	{
-		dbug(1, dprintf("[%06lx] %s,%d: LI peer in wrong state %08lx",
+		dbug(1, dprintf("[%06lx] %s,%d: LI peer in wrong state %08x",
 				UnMapId(Id), (char *)(FILE_), __LINE__, plci_b_id));
 		PUT_WORD(p_result, _WRONG_STATE);
 		return (NULL);
@@ -12009,7 +12009,7 @@ static PLCI *li2_check_plci_b(dword Id, PLCI *plci,
 	    && (!(plci->adapter->manufacturer_features & MANUFACTURER_FEATURE_XCONNECT)
 		|| !(plci_b->adapter->manufacturer_features & MANUFACTURER_FEATURE_XCONNECT)))
 	{
-		dbug(1, dprintf("[%06lx] %s,%d: LI not on same ctrl %08lx",
+		dbug(1, dprintf("[%06lx] %s,%d: LI not on same ctrl %08x",
 				UnMapId(Id), (char *)(FILE_), __LINE__, plci_b_id));
 		PUT_WORD(p_result, _WRONG_IDENTIFIER);
 		return (NULL);
@@ -12549,7 +12549,7 @@ static byte mixer_notify_source_removed(PLCI *plci, dword plci_b_id)
 	if (((plci->li_plci_b_read_pos > plci_b_write_pos) ? plci->li_plci_b_read_pos :
 	     LI_PLCI_B_QUEUE_ENTRIES + plci->li_plci_b_read_pos) - plci_b_write_pos - 1 < 1)
 	{
-		dbug(1, dprintf("[%06lx] %s,%d: LI request overrun",
+		dbug(1, dprintf("[%06x] %s,%d: LI request overrun",
 				(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 				(char *)(FILE_), __LINE__));
 		return (false);
@@ -12568,7 +12568,7 @@ static void mixer_remove(PLCI *plci)
 	dword plci_b_id;
 	word i, j;
 
-	dbug(1, dprintf("[%06lx] %s,%d: mixer_remove",
+	dbug(1, dprintf("[%06x] %s,%d: mixer_remove",
 			(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 			(char *)(FILE_), __LINE__));
 
@@ -12620,7 +12620,7 @@ static void ec_write_parameters(PLCI *plci)
 	word w;
 	byte parameter_buffer[6];
 
-	dbug(1, dprintf("[%06lx] %s,%d: ec_write_parameters",
+	dbug(1, dprintf("[%06x] %s,%d: ec_write_parameters",
 			(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 			(char *)(FILE_), __LINE__));
 
@@ -12639,7 +12639,7 @@ static void ec_write_parameters(PLCI *plci)
 static void ec_clear_config(PLCI *plci)
 {
 
-	dbug(1, dprintf("[%06lx] %s,%d: ec_clear_config",
+	dbug(1, dprintf("[%06x] %s,%d: ec_clear_config",
 			(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 			(char *)(FILE_), __LINE__));
 
@@ -13075,7 +13075,7 @@ static void adv_voice_write_coefs(PLCI *plci, word write_command)
 
 	byte coef_buffer[ADV_VOICE_COEF_BUFFER_SIZE + 2];
 
-	dbug(1, dprintf("[%06lx] %s,%d: adv_voice_write_coefs %d",
+	dbug(1, dprintf("[%06x] %s,%d: adv_voice_write_coefs %d",
 			(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 			(char *)(FILE_), __LINE__, write_command));
 
@@ -13102,7 +13102,7 @@ static void adv_voice_write_coefs(PLCI *plci, word write_command)
 		{
 			plci->li_bchannel_id = 1;
 			li_config_table[a->li_base].plci = plci;
-			dbug(1, dprintf("[%06lx] %s,%d: adv_voice_set_bchannel_id %d",
+			dbug(1, dprintf("[%06x] %s,%d: adv_voice_set_bchannel_id %d",
 					(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 					(char *)(FILE_), __LINE__, plci->li_bchannel_id));
 		}
@@ -13110,7 +13110,7 @@ static void adv_voice_write_coefs(PLCI *plci, word write_command)
 		{
 			plci->li_bchannel_id = 2;
 			li_config_table[a->li_base + 1].plci = plci;
-			dbug(1, dprintf("[%06lx] %s,%d: adv_voice_set_bchannel_id %d",
+			dbug(1, dprintf("[%06x] %s,%d: adv_voice_set_bchannel_id %d",
 					(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 					(char *)(FILE_), __LINE__, plci->li_bchannel_id));
 		}
@@ -13226,7 +13226,7 @@ static void adv_voice_clear_config(PLCI *plci)
 	word i, j;
 
 
-	dbug(1, dprintf("[%06lx] %s,%d: adv_voice_clear_config",
+	dbug(1, dprintf("[%06x] %s,%d: adv_voice_clear_config",
 			(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 			(char *)(FILE_), __LINE__));
 
@@ -13519,7 +13519,7 @@ static byte add_b1_facilities(PLCI *plci, byte b1_resource, word b1_facilities)
 	default:
 		b = b1_resource;
 	}
-	dbug(1, dprintf("[%06lx] %s,%d: add_b1_facilities %d %04x %d %04x",
+	dbug(1, dprintf("[%06x] %s,%d: add_b1_facilities %d %04x %d %04x",
 			(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 			(char *)(FILE_), __LINE__,
 			b1_resource, b1_facilities, b, get_b1_facilities(plci, b)));
@@ -13531,7 +13531,7 @@ static void adjust_b1_facilities(PLCI *plci, byte new_b1_resource, word new_b1_f
 {
 	word removed_facilities;
 
-	dbug(1, dprintf("[%06lx] %s,%d: adjust_b1_facilities %d %04x %04x",
+	dbug(1, dprintf("[%06x] %s,%d: adjust_b1_facilities %d %04x %04x",
 			(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 			(char *)(FILE_), __LINE__, new_b1_resource, new_b1_facilities,
 			new_b1_facilities & get_b1_facilities(plci, new_b1_resource)));
@@ -13564,7 +13564,7 @@ static void adjust_b1_facilities(PLCI *plci, byte new_b1_resource, word new_b1_f
 static void adjust_b_clear(PLCI *plci)
 {
 
-	dbug(1, dprintf("[%06lx] %s,%d: adjust_b_clear",
+	dbug(1, dprintf("[%06x] %s,%d: adjust_b_clear",
 			(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 			(char *)(FILE_), __LINE__));
 
@@ -14505,7 +14505,7 @@ static void retrieve_restore_command(dword Id, PLCI *plci, byte Rc)
 static void init_b1_config(PLCI *plci)
 {
 
-	dbug(1, dprintf("[%06lx] %s,%d: init_b1_config",
+	dbug(1, dprintf("[%06x] %s,%d: init_b1_config",
 			(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 			(char *)(FILE_), __LINE__));
 
@@ -14531,7 +14531,7 @@ static void init_b1_config(PLCI *plci)
 static void clear_b1_config(PLCI *plci)
 {
 
-	dbug(1, dprintf("[%06lx] %s,%d: clear_b1_config",
+	dbug(1, dprintf("[%06x] %s,%d: clear_b1_config",
 			(dword)((plci->Id << 8) | UnMapController(plci->adapter->Id)),
 			(char *)(FILE_), __LINE__));
 
@@ -14796,7 +14796,7 @@ static void group_optimization(DIVA_CAPI_ADAPTER *a, PLCI *plci)
 				{                                      /* is group already present ?                  */
 					appl_number_group_type[i] = j | 0x80;  /* store the group number for each application */
 					group_found = true;
-					dbug(1, dprintf("Group 0x%x found with appl 0x%x, CIP=0x%lx", appl_number_group_type[i], i + 1, info_mask_group[j]));
+					dbug(1, dprintf("Group 0x%x found with appl 0x%x, CIP=0x%x", appl_number_group_type[i], i + 1, info_mask_group[j]));
 				}
 				else if (!info_mask_group[j])
 				{                                      /* establish a new group                       */
@@ -14804,7 +14804,7 @@ static void group_optimization(DIVA_CAPI_ADAPTER *a, PLCI *plci)
 					info_mask_group[j] = a->CIP_Mask[i]; /* store the new CIP mask for the new group    */
 					cip_mask_group[j] = a->Info_Mask[i]; /* store the new Info_Mask for this new group  */
 					group_found = true;
-					dbug(1, dprintf("New Group 0x%x established with appl 0x%x, CIP=0x%lx", appl_number_group_type[i], i + 1, info_mask_group[j]));
+					dbug(1, dprintf("New Group 0x%x established with appl 0x%x, CIP=0x%x", appl_number_group_type[i], i + 1, info_mask_group[j]));
 				}
 			}
 		}
diff --git a/drivers/isdn/hardware/eicon/mntfunc.c b/drivers/isdn/hardware/eicon/mntfunc.c
index 1cd9affb6058..91bc8b93593c 100644
--- a/drivers/isdn/hardware/eicon/mntfunc.c
+++ b/drivers/isdn/hardware/eicon/mntfunc.c
@@ -34,7 +34,7 @@ extern int diva_os_copy_to_user(void *os_handle, void __user *dst,
 extern int diva_os_copy_from_user(void *os_handle, void *dst,
 				  const void __user *src, int length);
 
-static void no_printf(unsigned char *x, ...)
+static __printf(1, 2) void no_printf(const char *x, ...)
 {
 	/* dummy debug function */
 }
diff --git a/drivers/isdn/hardware/eicon/os_4bri.c b/drivers/isdn/hardware/eicon/os_4bri.c
index 1891246807ed..eb3a56c1b4f0 100644
--- a/drivers/isdn/hardware/eicon/os_4bri.c
+++ b/drivers/isdn/hardware/eicon/os_4bri.c
@@ -165,7 +165,7 @@ int diva_4bri_init_card(diva_os_xdi_adapter_t *a)
 	} else {
 		memcpy(bar_length, _4bri_bar_length, sizeof(bar_length));
 	}
-	DBG_TRC(("SDRAM_LENGTH=%08x, tasks=%d, factor=%d",
+	DBG_TRC(("SDRAM_LENGTH=%08lx, tasks=%d, factor=%d",
 		 bar_length[2], tasks, factor))
 
 		/*
@@ -184,7 +184,7 @@ int diva_4bri_init_card(diva_os_xdi_adapter_t *a)
 	  Set properties
 	*/
 	a->xdi_adapter.Properties = CardProperties[a->CardOrdinal];
-	DBG_LOG(("Load %s, SN:%ld, bus:%02x, func:%02x",
+	DBG_LOG(("Load %s, SN:%d, bus:%02x, func:%02x",
 		 a->xdi_adapter.Properties.Name,
 		 a->xdi_adapter.serialNo,
 		 a->resources.pci.bus, a->resources.pci.func))
@@ -555,7 +555,10 @@ static int _4bri_get_serial_number(diva_os_xdi_adapter_t *a)
 		}
 		PCIread(Bus, Slot, 0x50, &data[i], sizeof(data[i]), hdev);
 	}
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-security"
 	DBG_BLK(((char *) &data[0], sizeof(data)))
+#pragma GCC diagnostic pop
 
 		serNo = data[32];
 	if (serNo == 0 || serNo == 0xffffffff)
@@ -570,7 +573,7 @@ static int _4bri_get_serial_number(diva_os_xdi_adapter_t *a)
 
 	a->xdi_adapter.serialNo = serNo;
 
-	DBG_REG(("Serial No.          : %ld", a->xdi_adapter.serialNo))
+	DBG_REG(("Serial No.          : %d", a->xdi_adapter.serialNo))
 
 		return (serNo);
 }
@@ -915,7 +918,7 @@ diva_4bri_write_sdram_block(PISDN_ADAPTER IoAdapter,
 
 	if (((address + length) >= limit) || !mem) {
 		DIVA_OS_MEM_DETACH_ADDRESS(IoAdapter, p);
-		DBG_ERR(("A: A(%d) write 4BRI address=0x%08lx",
+		DBG_ERR(("A: A(%d) write 4BRI address=0x%08x",
 			 IoAdapter->ANum, address + length))
 			return (-1);
 	}
diff --git a/drivers/isdn/hardware/eicon/os_bri.c b/drivers/isdn/hardware/eicon/os_bri.c
index 20f2653c58fa..267b64bd2feb 100644
--- a/drivers/isdn/hardware/eicon/os_bri.c
+++ b/drivers/isdn/hardware/eicon/os_bri.c
@@ -139,7 +139,7 @@ int diva_bri_init_card(diva_os_xdi_adapter_t *a)
 	  Calculate BAR2 length
 	*/
 	bar2_length = (~(bar2_length & ~7)) + 1;
-	DBG_LOG(("BAR[2] length=%lx", bar2_length))
+	DBG_LOG(("BAR[2] length=%x", bar2_length))
 
 		/*
 		  Map and register resources
@@ -352,7 +352,7 @@ static dword diva_bri_get_serial_number(diva_os_xdi_adapter_t *a)
 		DIVA_OS_MEM_DETACH_RAM(&a->xdi_adapter, confMem);
 	}
 
-	DBG_LOG(("Serial Number=%ld", serNo))
+	DBG_LOG(("Serial Number=%d", serNo))
 
 		return (serNo);
 }
diff --git a/drivers/isdn/hardware/eicon/os_pri.c b/drivers/isdn/hardware/eicon/os_pri.c
index da4957abb422..d00fcf7e53a5 100644
--- a/drivers/isdn/hardware/eicon/os_pri.c
+++ b/drivers/isdn/hardware/eicon/os_pri.c
@@ -169,7 +169,7 @@ int diva_pri_init_card(diva_os_xdi_adapter_t *a)
 		serNo |= ((dword) a->resources.pci.bus) << 8;
 		serNo += (a->resources.pci.func + a->controller + 1);
 		a->xdi_adapter.serialNo = serNo & ~0xFF000000;
-		DBG_ERR(("A: A(%d) can't get Serial Number, generated serNo=%ld",
+		DBG_ERR(("A: A(%d) can't get Serial Number, generated serNo=%d",
 			 a->controller, a->xdi_adapter.serialNo))
 			}
 
@@ -358,12 +358,12 @@ static int diva_pri_reset_adapter(PISDN_ADAPTER IoAdapter)
 	diva_os_wait(10);
 	if (i == READ_DWORD(&boot->live)) {
 		DIVA_OS_MEM_DETACH_ADDRESS(IoAdapter, boot);
-		DBG_ERR(("A: A(%d) CPU on PRI %ld is not alive!",
+		DBG_ERR(("A: A(%d) CPU on PRI %d is not alive!",
 			 IoAdapter->ANum, IoAdapter->serialNo))
 			return (-1);
 	}
 	if (READ_DWORD(&boot->err)) {
-		DBG_ERR(("A: A(%d) PRI %ld Board Selftest failed, error=%08lx",
+		DBG_ERR(("A: A(%d) PRI %d Board Selftest failed, error=%08x",
 			 IoAdapter->ANum, IoAdapter->serialNo,
 			 READ_DWORD(&boot->err)))
 			DIVA_OS_MEM_DETACH_ADDRESS(IoAdapter, boot);
@@ -414,7 +414,7 @@ diva_pri_write_sdram_block(PISDN_ADAPTER IoAdapter,
 
 	if (((address + length) >= limit) || !mem) {
 		DIVA_OS_MEM_DETACH_ADDRESS(IoAdapter, p);
-		DBG_ERR(("A: A(%d) write PRI address=0x%08lx",
+		DBG_ERR(("A: A(%d) write PRI address=0x%08x",
 			 IoAdapter->ANum, address + length))
 			return (-1);
 	}
@@ -447,13 +447,13 @@ diva_pri_start_adapter(PISDN_ADAPTER IoAdapter,
 	}
 	if (!boot) {
 		DIVA_OS_MEM_DETACH_ADDRESS(IoAdapter, boot);
-		DBG_ERR(("A: PRI %ld can't start, adapter not mapped",
+		DBG_ERR(("A: PRI %d can't start, adapter not mapped",
 			 IoAdapter->serialNo))
 			return (-1);
 	}
 
 	sprintf(IoAdapter->Name, "A(%d)", (int) IoAdapter->ANum);
-	DBG_LOG(("A(%d) start PRI at 0x%08lx", IoAdapter->ANum,
+	DBG_LOG(("A(%d) start PRI at 0x%08x", IoAdapter->ANum,
 		 start_address))
 
 		WRITE_DWORD(&boot->addr, start_address);
@@ -475,7 +475,7 @@ diva_pri_start_adapter(PISDN_ADAPTER IoAdapter,
 		dword debug;
 		TrapId = READ_DWORD(&p[0x80]);
 		debug = READ_DWORD(&p[0x1c]);
-		DBG_ERR(("A(%d) Adapter start failed 0x%08lx, TrapId=%08lx, debug=%08lx",
+		DBG_ERR(("A(%d) Adapter start failed 0x%08x, TrapId=%08x, debug=%08x",
 			 IoAdapter->ANum, READ_DWORD(&boot->signature),
 			 TrapId, debug))
 			DIVA_OS_MEM_DETACH_ADDRESS(IoAdapter, boot);
@@ -882,19 +882,19 @@ static int pri_get_serial_number(diva_os_xdi_adapter_t *a)
 			return (-1);
 	}
 
-	DBG_LOG(("Serial No.          : %ld", a->xdi_adapter.serialNo))
-		DBG_TRC(("Board Revision      : %d.%02d", (int) data[41],
-			 (int) data[40]))
-		DBG_TRC(("PLD revision        : %d.%02d", (int) data[33],
-			 (int) data[32]))
-		DBG_TRC(("Boot loader version : %d.%02d", (int) data[37],
-			 (int) data[36]))
+	DBG_LOG(("Serial No.          : %d", a->xdi_adapter.serialNo))
+	DBG_TRC(("Board Revision      : %d.%02d", (int) data[41],
+		 (int) data[40]))
+	DBG_TRC(("PLD revision        : %d.%02d", (int) data[33],
+		 (int) data[32]))
+	DBG_TRC(("Boot loader version : %d.%02d", (int) data[37],
+		 (int) data[36]))
 
-		DBG_TRC(("Manufacturing Date  : %d/%02d/%02d  (yyyy/mm/dd)",
-			 (int) ((data[28] > 90) ? 1900 : 2000) +
-			 (int) data[28], (int) data[29], (int) data[30]))
+	DBG_TRC(("Manufacturing Date  : %d/%02d/%02d  (yyyy/mm/dd)",
+		 (int) ((data[28] > 90) ? 1900 : 2000) +
+		 (int) data[28], (int) data[29], (int) data[30]))
 
-		return (0);
+	return (0);
 }
 
 void diva_os_prepare_pri2_functions(PISDN_ADAPTER IoAdapter)
diff --git a/drivers/isdn/hardware/eicon/platform.h b/drivers/isdn/hardware/eicon/platform.h
index f1faa55b2ef6..91766029c2fb 100644
--- a/drivers/isdn/hardware/eicon/platform.h
+++ b/drivers/isdn/hardware/eicon/platform.h
@@ -132,7 +132,7 @@ typedef char DEVICE_NAME[64];
 typedef struct _ISDN_ADAPTER ISDN_ADAPTER;
 typedef struct _ISDN_ADAPTER *PISDN_ADAPTER;
 
-typedef void (*DIVA_DI_PRINTF)(unsigned char *, ...);
+typedef __printf(1, 2) void (*DIVA_DI_PRINTF)(const char *, ...);
 #include "debuglib.h"
 
 #define dtrc(p) DBG_PRV0(p)
@@ -148,7 +148,7 @@ struct _diva_os_soft_isr;
 struct _diva_os_timer;
 struct _ISDN_ADAPTER;
 
-void diva_log_info(unsigned char *, ...);
+__printf(1, 2) void diva_log_info(const char *, ...);
 
 /*
 **  XDI DIDD Interface
diff --git a/drivers/isdn/hardware/eicon/s_4bri.c b/drivers/isdn/hardware/eicon/s_4bri.c
index ec12165fbf62..4fd5903e2ba8 100644
--- a/drivers/isdn/hardware/eicon/s_4bri.c
+++ b/drivers/isdn/hardware/eicon/s_4bri.c
@@ -118,9 +118,9 @@ static void reset_qBri_hardware(PISDN_ADAPTER IoAdapter) {
 	WRITE_DWORD(p, 0);
 	DIVA_OS_MEM_DETACH_CTLREG(IoAdapter, qBriCntrl);
 
-	DBG_TRC(("resetted board @ reset addr 0x%08lx", qBriReset))
-		DBG_TRC(("resetted board @ cntrl addr 0x%08lx", p))
-		}
+	DBG_TRC(("resetted board @ reset addr %p", qBriReset))
+	DBG_TRC(("resetted board @ cntrl addr %p", p))
+}
 
 /* --------------------------------------------------------------------------
    Start Card CPU
@@ -137,8 +137,8 @@ void start_qBri_hardware(PISDN_ADAPTER IoAdapter) {
 	diva_os_wait(10);
 	DIVA_OS_MEM_DETACH_CTLREG(IoAdapter, p);
 
-	DBG_TRC(("started processor @ addr 0x%08lx", qBriReset))
-		}
+	DBG_TRC(("started processor @ addr %p", qBriReset))
+}
 
 /* --------------------------------------------------------------------------
    Stop Card CPU
@@ -174,9 +174,9 @@ static void stop_qBri_hardware(PISDN_ADAPTER IoAdapter) {
 	WRITE_DWORD(qBriIrq, MQ_IRQ_REQ_OFF);
 	DIVA_OS_MEM_DETACH_CTLREG(IoAdapter, p);
 
-	DBG_TRC(("stopped processor @ addr 0x%08lx", qBriReset))
+	DBG_TRC(("stopped processor @ addr %p", qBriReset))
 
-		}
+}
 
 /* --------------------------------------------------------------------------
    FPGA download
@@ -230,8 +230,8 @@ static byte *qBri_check_FPGAsrc(PISDN_ADAPTER IoAdapter, char *FileName,
 
 	if ((dword)(i + (cnt / 8)) > *Length)
 	{
-		DBG_FTL(("FPGA download: '%s' file too small (%ld < %ld)",
-			 FileName, *Length, code + ((cnt + 7) / 8)))
+		DBG_FTL(("FPGA download: '%s' file too small (%d < %d)",
+			 FileName, *Length, *code + ((cnt + 7) / 8)))
 			xdiFreeFile(File);
 		return (NULL);
 	}
diff --git a/drivers/isdn/hardware/eicon/um_idi.c b/drivers/isdn/hardware/eicon/um_idi.c
index 13ef38fa6cb0..a4dc3439ef44 100644
--- a/drivers/isdn/hardware/eicon/um_idi.c
+++ b/drivers/isdn/hardware/eicon/um_idi.c
@@ -216,7 +216,7 @@ void *divas_um_idi_create_entity(dword adapter_nr, void *file)
 		if (!
 		    (e->os_context =
 		     diva_os_malloc(0, diva_os_get_context_size()))) {
-			DBG_LOG(("E(%08x) no memory for os context", e));
+			DBG_LOG(("E(%p) no memory for os context", e));
 			diva_os_free(0, e);
 			return NULL;
 		}
@@ -244,7 +244,7 @@ void *divas_um_idi_create_entity(dword adapter_nr, void *file)
 			*/
 			diva_os_leave_spin_lock(&adapter_lock, &old_irql, "create_entity");
 
-			DBG_LOG(("A: no adapter(%ld)", adapter_nr));
+			DBG_LOG(("A: no adapter(%d)", adapter_nr));
 
 			cleanup_entity(e);
 			diva_os_free(0, e->os_context);
@@ -260,7 +260,7 @@ void *divas_um_idi_create_entity(dword adapter_nr, void *file)
 
 		diva_os_leave_spin_lock(&adapter_lock, &old_irql, "create_entity");
 
-		DBG_LOG(("A(%ld), create E(%08x)", adapter_nr, e));
+		DBG_LOG(("A(%d), create E(%p)", adapter_nr, e));
 	}
 
 	return (e);
@@ -289,7 +289,7 @@ int divas_um_idi_delete_entity(int adapter_nr, void *entity)
 	diva_os_free(0, e->os_context);
 	memset(e, 0x00, sizeof(*e));
 
-	DBG_LOG(("A(%d) remove E:%08x", adapter_nr, e));
+	DBG_LOG(("A(%d) remove E:%p", adapter_nr, e));
 	diva_os_free(0, e);
 
 	return (0);
@@ -318,11 +318,11 @@ int diva_um_idi_read(void *entity,
 	    (e->status & DIVA_UM_IDI_REMOVED) ||
 	    (a->status & DIVA_UM_IDI_ADAPTER_REMOVED)) {
 		diva_os_leave_spin_lock(&adapter_lock, &old_irql, "read");
-		DBG_ERR(("E(%08x) read failed - adapter removed", e))
+		DBG_ERR(("E(%p) read failed - adapter removed", e))
 			return (-1);
 	}
 
-	DBG_TRC(("A(%d) E(%08x) read(%d)", a->adapter_nr, e, max_length));
+	DBG_TRC(("A(%d) E(%p) read(%d)", a->adapter_nr, e, max_length));
 
 	/*
 	  Try to read return code first
@@ -335,13 +335,13 @@ int diva_um_idi_read(void *entity,
 	*/
 	if (!data) {
 		if (!(e->status & DIVA_UM_IDI_RC_PENDING)) {
-			DBG_TRC(("A(%d) E(%08x) read data", a->adapter_nr, e));
+			DBG_TRC(("A(%d) E(%p) read data", a->adapter_nr, e));
 			data = diva_data_q_get_segment4read(&e->data);
 			q = &e->data;
 		}
 	} else {
 		e->status &= ~DIVA_UM_IDI_RC_PENDING;
-		DBG_TRC(("A(%d) E(%08x) read rc", a->adapter_nr, e));
+		DBG_TRC(("A(%d) E(%p) read rc", a->adapter_nr, e));
 	}
 
 	if (data) {
@@ -350,7 +350,7 @@ int diva_um_idi_read(void *entity,
 			/*
 			  Not enough space to read message
 			*/
-			DBG_ERR(("A: A(%d) E(%08x) read small buffer",
+			DBG_ERR(("A: A(%d) E(%p) read small buffer",
 				 a->adapter_nr, e));
 			diva_os_leave_spin_lock(&adapter_lock, &old_irql,
 						"read");
@@ -370,7 +370,7 @@ int diva_um_idi_read(void *entity,
 	}
 
 
-	DBG_TRC(("A(%d) E(%08x) read=%d", a->adapter_nr, e, ret));
+	DBG_TRC(("A(%d) E(%p) read=%d", a->adapter_nr, e, ret));
 
 	diva_os_leave_spin_lock(&adapter_lock, &old_irql, "read");
 
@@ -398,11 +398,11 @@ int diva_um_idi_write(void *entity,
 	    (e->status & DIVA_UM_IDI_REMOVED) ||
 	    (a->status & DIVA_UM_IDI_ADAPTER_REMOVED)) {
 		diva_os_leave_spin_lock(&adapter_lock, &old_irql, "write");
-		DBG_ERR(("E(%08x) write failed - adapter removed", e))
+		DBG_ERR(("E(%p) write failed - adapter removed", e))
 			return (-1);
 	}
 
-	DBG_TRC(("A(%d) E(%08x) write(%d)", a->adapter_nr, e, length));
+	DBG_TRC(("A(%d) E(%p) write(%d)", a->adapter_nr, e, length));
 
 	if ((length < sizeof(*req)) || (length > sizeof(e->buffer))) {
 		diva_os_leave_spin_lock(&adapter_lock, &old_irql, "write");
@@ -410,7 +410,7 @@ int diva_um_idi_write(void *entity,
 	}
 
 	if (e->status & DIVA_UM_IDI_RC_PENDING) {
-		DBG_ERR(("A: A(%d) E(%08x) rc pending", a->adapter_nr, e));
+		DBG_ERR(("A: A(%d) E(%p) rc pending", a->adapter_nr, e));
 		diva_os_leave_spin_lock(&adapter_lock, &old_irql, "write");
 		return (-1);	/* should wait for RC code first */
 	}
@@ -420,7 +420,7 @@ int diva_um_idi_write(void *entity,
 	  also it can be called with spin lock held
 	*/
 	if ((ret = (*cp_fn) (os_handle, e->buffer, src, length)) < 0) {
-		DBG_TRC(("A: A(%d) E(%08x) write error=%d", a->adapter_nr,
+		DBG_TRC(("A: A(%d) E(%p) write error=%d", a->adapter_nr,
 			 e, ret));
 		diva_os_leave_spin_lock(&adapter_lock, &old_irql, "write");
 		return (ret);
@@ -480,7 +480,7 @@ int diva_um_idi_write(void *entity,
 		return (-1);
 	}
 
-	DBG_TRC(("A(%d) E(%08x) write=%d", a->adapter_nr, e, ret));
+	DBG_TRC(("A(%d) E(%p) write=%d", a->adapter_nr, e, ret));
 
 	return (ret);
 }
@@ -538,31 +538,31 @@ static int process_idi_request(divas_um_idi_entity_t *e,
 
 	if (!e->e.Id || !e->e.callback) {	/* not assigned */
 		if (Req != ASSIGN) {
-			DBG_ERR(("A: A(%d) E(%08x) not assigned",
+			DBG_ERR(("A: A(%d) E(%p) not assigned",
 				 e->adapter->adapter_nr, e));
 			return (-1);	/* NOT ASSIGNED */
 		} else {
 			switch (type) {
 			case DIVA_UM_IDI_REQ_TYPE_MAN:
 				e->e.Id = MAN_ID;
-				DBG_TRC(("A(%d) E(%08x) assign MAN",
+				DBG_TRC(("A(%d) E(%p) assign MAN",
 					 e->adapter->adapter_nr, e));
 				break;
 
 			case DIVA_UM_IDI_REQ_TYPE_SIG:
 				e->e.Id = DSIG_ID;
-				DBG_TRC(("A(%d) E(%08x) assign SIG",
+				DBG_TRC(("A(%d) E(%p) assign SIG",
 					 e->adapter->adapter_nr, e));
 				break;
 
 			case DIVA_UM_IDI_REQ_TYPE_NET:
 				e->e.Id = NL_ID;
-				DBG_TRC(("A(%d) E(%08x) assign NET",
+				DBG_TRC(("A(%d) E(%p) assign NET",
 					 e->adapter->adapter_nr, e));
 				break;
 
 			default:
-				DBG_ERR(("A: A(%d) E(%08x) unknown type=%08x",
+				DBG_ERR(("A: A(%d) E(%p) unknown type=%08x",
 					 e->adapter->adapter_nr, e,
 					 type));
 				return (-1);
@@ -581,7 +581,7 @@ static int process_idi_request(divas_um_idi_entity_t *e,
 	e->e.X->PLength = (word) req->data_length;
 	e->e.X->P = (byte *)&req[1];	/* Our buffer is safe */
 
-	DBG_TRC(("A(%d) E(%08x) request(%02x-%02x-%02x (%d))",
+	DBG_TRC(("A(%d) E(%p) request(%02x-%02x-%02x (%d))",
 		 e->adapter->adapter_nr, e, e->e.Id, e->e.Req,
 		 e->e.ReqCh, e->e.X->PLength));
 
@@ -598,7 +598,7 @@ static int process_idi_request(divas_um_idi_entity_t *e,
 			  XDI has no entities more, call was not forwarded to the card,
 			  no callback will be scheduled
 			*/
-			DBG_ERR(("A: A(%d) E(%08x) XDI out of entities",
+			DBG_ERR(("A: A(%d) E(%p) XDI out of entities",
 				 e->adapter->adapter_nr, e));
 
 			e->e.Id = 0;
@@ -623,13 +623,13 @@ static int process_idi_request(divas_um_idi_entity_t *e,
 
 static int process_idi_rc(divas_um_idi_entity_t *e, byte rc)
 {
-	DBG_TRC(("A(%d) E(%08x) rc(%02x-%02x-%02x)",
+	DBG_TRC(("A(%d) E(%p) rc(%02x-%02x-%02x)",
 		 e->adapter->adapter_nr, e, e->e.Id, rc, e->e.RcCh));
 
 	if (e->status & DIVA_UM_IDI_ASSIGN_PENDING) {
 		e->status &= ~DIVA_UM_IDI_ASSIGN_PENDING;
 		if (rc != ASSIGN_OK) {
-			DBG_ERR(("A: A(%d) E(%08x) ASSIGN failed",
+			DBG_ERR(("A: A(%d) E(%p) ASSIGN failed",
 				 e->adapter->adapter_nr, e));
 			e->e.callback = NULL;
 			e->e.Id = 0;
@@ -646,7 +646,7 @@ static int process_idi_rc(divas_um_idi_entity_t *e, byte rc)
 		}
 	}
 	if ((e->e.Req == REMOVE) && e->e.Id && (rc == 0xff)) {
-		DBG_ERR(("A: A(%d) E(%08x)  discard OK in REMOVE",
+		DBG_ERR(("A: A(%d) E(%p)  discard OK in REMOVE",
 			 e->adapter->adapter_nr, e));
 		return (0);	/* let us do it in the driver */
 	}
@@ -666,7 +666,7 @@ static int process_idi_rc(divas_um_idi_entity_t *e, byte rc)
 		e->rc_count = 0;
 	}
 	if ((e->e.Req == REMOVE) && (rc != 0xff)) {	/* REMOVE FAILED */
-		DBG_ERR(("A: A(%d) E(%08x)  REMOVE FAILED",
+		DBG_ERR(("A: A(%d) E(%p)  REMOVE FAILED",
 			 e->adapter->adapter_nr, e));
 	}
 	write_return_code(e, rc);
@@ -688,13 +688,13 @@ static int process_idi_ind(divas_um_idi_entity_t *e, byte ind)
 			e->e.R->PLength =
 				(word) (diva_data_q_get_max_length(&e->data) -
 					sizeof(*pind));
-			DBG_TRC(("A(%d) E(%08x) ind_1(%02x-%02x-%02x)-[%d-%d]",
+			DBG_TRC(("A(%d) E(%p) ind_1(%02x-%02x-%02x)-[%d-%d]",
 				 e->adapter->adapter_nr, e, e->e.Id, ind,
 				 e->e.IndCh, e->e.RLength,
 				 e->e.R->PLength));
 
 		} else {
-			DBG_TRC(("A(%d) E(%08x) ind(%02x-%02x-%02x)-RNR",
+			DBG_TRC(("A(%d) E(%p) ind(%02x-%02x-%02x)-RNR",
 				 e->adapter->adapter_nr, e, e->e.Id, ind,
 				 e->e.IndCh));
 			e->e.RNum = 0;
@@ -705,7 +705,7 @@ static int process_idi_ind(divas_um_idi_entity_t *e, byte ind)
 		diva_um_idi_ind_hdr_t *pind =
 			(diva_um_idi_ind_hdr_t *) (e->e.R->P);
 
-		DBG_TRC(("A(%d) E(%08x) ind(%02x-%02x-%02x)-[%d]",
+		DBG_TRC(("A(%d) E(%p) ind(%02x-%02x-%02x)-[%d]",
 			 e->adapter->adapter_nr, e, e->e.Id, ind,
 			 e->e.IndCh, e->e.R->PLength));
 
@@ -737,7 +737,7 @@ static int write_return_code(divas_um_idi_entity_t *e, byte rc)
 	if (!(prc =
 	      (diva_um_idi_ind_hdr_t *) diva_data_q_get_segment4write(&e->rc)))
 	{
-		DBG_ERR(("A: A(%d) E(%08x) rc(%02x) lost",
+		DBG_ERR(("A: A(%d) E(%p) rc(%02x) lost",
 			 e->adapter->adapter_nr, e, rc));
 		e->status &= ~DIVA_UM_IDI_RC_PENDING;
 		return (-1);
@@ -872,7 +872,7 @@ int divas_um_idi_entity_start_remove(void *entity)
 
 	e->rc_count++;
 
-	DBG_TRC(("A(%d) E(%08x) request(%02x-%02x-%02x (%d))",
+	DBG_TRC(("A(%d) E(%p) request(%02x-%02x-%02x (%d))",
 		 e->adapter->adapter_nr, e, e->e.Id, e->e.Req,
 		 e->e.ReqCh, e->e.X->PLength));
 
-- 
